StrangeIoC/scripts/strange/extensions/mediation/impl/MediationBinder.cs

```diff
@@ -17,212 +17,87 @@
-			{
-				//Even if not mapped, Views (and their children) have potential to be injected
-				injectViewAndChildren(view);
-			}
-		}
-
-		override protected IBinding performKeyValueBindings(List<object> keyList, List<object> valueList)
-		{
-			IBinding binding = null;
-
-			// Bind in order
-			foreach (object key in keyList)
-			{
-				Type keyType = Type.GetType (key as string);
-				if (keyType == null)
-				{
-					throw new BinderException ("A runtime Mediation Binding has resolved to null. Did you forget to register its fully-qualified name?\n View:" + key, BinderExceptionType.RUNTIME_NULL_VALUE);
-				}
-				if (binding == null)
-				{
-					binding = Bind (keyType);
-				}
-				else
-				{
-					binding = binding.Bind (keyType);
-				}
-			}
-			foreach (object value in valueList)
-			{
-				Type valueType = Type.GetType (value as string);
-				if (valueType == null)
-				{
-					throw new BinderException ("A runtime Mediation Binding has resolved to null. Did you forget to register its fully-qualified name?\n Mediator:" + value, BinderExceptionType.RUNTIME_NULL_VALUE);
-				}
-				binding = binding.To (valueType);
-			}
-
-			return binding;
-		}
 		
 		/// Initialize all IViews within this view
-		virtual protected void injectViewAndChildren(IView view)
+		protected override void InjectViewAndChildren(IView view)
 		{
 			MonoBehaviour mono = view as MonoBehaviour;
-			Component[] views = mono.GetComponentsInChildren(typeof(IView), true) as Component[];
+			IView[] views = mono.GetComponentsInChildren(typeof(IView), true) as IView[];
 			
-			int aa = views.Length;
-			for (int a = aa - 1; a > -1; a--)
-			{
-				IView iView = views[a] as IView;
-				if (iView != null)
-				{
-					if (iView.autoRegisterWithContext && iView.registeredWithContext)
-					{
-						continue;
-					}
-					iView.registeredWithContext = true;
-					if (iView.Equals(mono) == false)
-						Trigger (MediationEvent.AWAKE, iView);
-				}
-			}
-			injectionBinder.injector.Inject (mono, false);
-
-		}
-
-		new public IMediationBinding Bind<T> ()
-		{
-			return base.Bind<T> () as IMediationBinding;
+			InjectViews(mono, views);
 		}
 
-		public IMediationBinding BindView<T>() where T : MonoBehaviour
+		/// Add a Mediator to a View. If the mediator is a "true" Mediator (i.e., it
+		/// implements IMediator, perform PreRegister and OnRegister.
+		protected override void ApplyMediationToView(IMediationBinding binding, IView view, Type mediatorType)
 		{
-			return base.Bind<T> () as IMediationBinding;
-		}
-
-		/// Creates and registers one or more Mediators for a specific View instance.
-		/// Takes a specific View instance and a binding and, if a binding is found for that type, creates and registers a Mediator.
-		virtual protected void mapView(IView view, IMediationBinding binding)
-		{
-			Type viewType = view.GetType();
-
-			if (bindings.ContainsKey(viewType))
+			bool isTrueMediator = mediatorType.IsAssignableFrom (typeof(IMediator));
```

Comment:
This looks backwards. Fixing on my local branch. Just thought I'd point it out for future use. interfaceType.isassignablefrom(myType)